### DISCUSSION TITLE
API Script for Candidate De-dupe

### DISCUSSION THREAD

**Discussion by CHRISTIEKIMBALL** [bullhorn/career-portal#423] [Q&A]
Hi All -  How would a client use APIs to write code to replicate the candidate merge function in Bullhorn? Their end goal is to deduplicate their candidate records in the database. Is this the best way to complete this? Thank you!

**Comment by billhess** [comment]
I assume you are referring to the Bullhorn REST APIs and/or the older SOAP
APIs

You can get to pretty much whatever info you want for candidates, jobs etc
You would have to determine what logic to use to flag a duplicate candidate
- comparing names will not be good enough - so you would cross reference
email addresses and even phone numbers to determine a duplicate.  The
searching in Bullhorn via the API can be expensive in terms of
time/resources - so depending how often youre going to check for dups - you
may need to get creative and develop a scheme to help determine dups more
efficiently.

We have have a subscription set up to flag any candidate records that have
changed and have programs that extract some info out of those records and
pump that info into our own internal database where we can do dup checks
much more efficiently and report back to us when it determines dups and we
can go and merge them easily

Bill Hess
TRG - 3001 West Big Beaver Road - Suite 720 - Troy, MI 48084
mobile 248.703.1196  |  office 248.817.4403   |   techrg.com

**Reply by pbearne** [reply]
It might be worth you talking to Kyloe https://kyloepartners.com/ as they have ready-made tools
